src/pkg/caendr/caendr/models/datastore/cart.py

```diff
@@ -35,29 +32,30 @@ def get_props_set(cls):
       *super().get_props_set(),
       'user',
       'items',
-      'is_deleted',
       'version'
     }
   
   
 
   @classmethod
   def lookup_by_user(cls, email):
-
-    # Try finding and returning unique card for this user
+    
+    # Try finding cart(s) for user
     try:
-      return Cart.query_ds_not_deleted('user', email, required=True)
+      # Run the query and try to find not deleted cart(s) for the user
+      carts = cls.query_ds_not_deleted('user', email, required=True)
 
+      if len(carts) == 1:
```

Comment:
A little trick for this part -- if you walk through the logic, the `else` branch will actually work in both cases!  (Since sorting a single-element list doesn't do anything to it.)  If you collapse this `if-else` statement into a single line (just line 53), you end up with a single line of code and with fewer edge cases.

If you decide to go this route, you'd probably want to adjust the comment to make it clear what you're doing.

Reply:
Good catch! Removed if/else condition in the commit [b066433](https://github.com/AndersenLab/CAENDR/pull/403/commits/b066433938e167756db66dc9c06f9bb7a1b330cd)